app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageItemComponents.kt

```diff
@@ -80,54 +84,97 @@ internal fun MessageSentPartialDeliveryFailures(partialDeliveryFailureContent: D
     CompositionLocalProvider(
         LocalTextStyle provides MaterialTheme.typography.labelSmall.copy(color = MaterialTheme.wireColorScheme.error)
     ) {
-        Column {
-            Text(
-                text = stringResource(
-                    id = R.string.label_message_partial_delivery_participants_count,
-                    partialDeliveryFailureContent.totalUsersWithFailures
-                ),
-                textAlign = TextAlign.Start
-            )
-            VerticalSpace.x4()
-            if (expanded) {
-                if (partialDeliveryFailureContent.noClients.isNotEmpty()) {
-                    Text(
-                        text = stringResource(
-                            id = R.string.label_message_partial_delivery_participants_wont_deliver,
-                            partialDeliveryFailureContent.noClients.joinToString(", ") { it.asString(resources) }
-                        ),
-                        textAlign = TextAlign.Start
-                    )
+        if (partialDeliveryFailureContent.isSingleUserFailure) {
+            SingleUserDeliveryFailure(partialDeliveryFailureContent, resources)
+        } else {
+            Column {
```

Comment:
Can the whole `else` block be extracted to a `MultiUserDeliveryFailure` composable?

Reducing this function's responsibility and increasing readability by cutting the nesting.